rfdetr/detr.py

```diff
@@ -39,11 +40,32 @@ def export(self, **kwargs):
         self.model.export(**kwargs)
 
     def train_from_config(self, config: TrainConfig, **kwargs):
-        with open(
-            os.path.join(config.dataset_dir, "train", "_annotations.coco.json"), "r"
-        ) as f:
-            anns = json.load(f)
-            num_classes = len(anns["categories"])
+        # Check if dataset is in YOLO format
+        if is_valid_yolo_format(config.dataset_dir):
+            logger.info(f"Using native YOLO dataloader for dataset: {config.dataset_dir}")
+            
+            # Read number of classes from data.yaml
+            data_yaml_path = os.path.join(config.dataset_dir, "data.yaml")
+            import yaml
```

Comment:
let's create a separate `load_yaml` function, place it in `rfdetr/util/files.py` and move there snippet bellow:

```python
import yaml
with open(data_yaml_path, 'r') as f:
    data = yaml.safe_load(f)
``` 

Please make sure to include any necessary checks. Make sure util is general. Applicable to load any YAML file. ;)